osmoutils/accum/accum.go

```diff
@@ -0,0 +1,98 @@
+package accum
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/gogo/protobuf/proto"
+)
+
+// We keep this object as a way to interface with the methods, even though
+// only the Accumulator inside is stored in state
+type AccumulatorObject struct {
+	// Store where accumulator is stored
+	Store store.KVStore
+
+	// Accumulator's name (pulled from AccumulatorContent)
+	Name string
```

Comment:
We might want to lowercase these for better information hiding

Reply:
I see where you're coming from – is there a benefit to information hiding over the convenience (and potential utility) of being able to access e.g. the name/value of the accumulator in the caller? Having them public also simplifies testing a bit although that's a smaller issue.

Reply:
We can define exported getters for `Name` and `Value`.

I think that the main benefit of information hiding here is to prevent accidental modification. If users can only access these fields via getters, there is no way to mistakenly alter the accumulator value.

Also, I think that `Store` is better to be inaccessible from a security standpoint as well. One shouldn't really need to access the store outside of the internal accumulator logic

Reply:
Agreed (and discussed on call) – done!